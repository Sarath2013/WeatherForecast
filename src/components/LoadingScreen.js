import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';

const LoadingScreen = () => {
  return (
    <div className="loader-content center-content">
      <div className="loader-text">Loading ...</div>
      <CircularProgress />
    </div>
  );
}

export default LoadingScreen;