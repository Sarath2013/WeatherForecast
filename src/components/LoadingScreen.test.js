import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoadingScreen from './LoadingScreen';

let wrapper;

describe("Unit Testing - Rendering Loading Screen", () => {
    beforeAll(() => {
    });

    afterAll(() => {
        wrapper = null
    });

    test('renders snapshot', () => {
        wrapper = shallow(<LoadingScreen />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('ensure loading screen content', () => {
        expect(wrapper.find('.loader-text').text()).toEqual('Loading ...');
        expect(wrapper.props().className).toEqual('loader-content center-content');
    });
});