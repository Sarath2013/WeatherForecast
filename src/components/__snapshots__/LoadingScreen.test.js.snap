// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing - Rendering Loading Screen renders snapshot 1`] = `
<div
  className="loader-content center-content"
>
  <div
    className="loader-text"
  >
    Loading ...
  </div>
  <WithStyles(ForwardRef(CircularProgress)) />
</div>
`;
